pingora-core/src/server/mod.rs

```diff
@@ -59,6 +60,67 @@ pub type ShutdownWatch = watch::Receiver<bool>;
 #[cfg(unix)]
 pub type ListenFds = Arc<Mutex<Fds>>;
 
+/// The type of shutdown process that has been requested.
+pub enum ShutdownSignal {
+    /// Send file descriptors to the new process before starting runtime shutdown with
+    /// [ServerConf::graceful_shutdown_timeout_seconds] timeout.
+    GracefulUpgrade,
+    /// Wait for [ServerConf::grace_period_seconds] before starting runtime shutdown with
+    /// [ServerConf::graceful_shutdown_timeout_seconds] timeout.
+    GracefulTerminate,
+    /// Shutdown with no timeout for runtime shutdown.
+    FastShutdown,
+}
+
```

Comment:
Could you derive `Debug` here as well?

Reply:
Added `Debug` derive. Not sure if you also want me to go ahead and add to other related ones like `ShutdownType`?

Reply:
Note: I'm not affiliated with Cloudflare, I just started using your changes from here and ran into an issue due to the missing Debug impl.

Reply:
What is the issue you encountered?